### PR TITLE
UTF-8 protected

### PR DESCRIPTION
**Pull request by davidcarlisle**
This PR uses `\protected` definitions for the UTF-8 buyes set up in `uf8.def` for UTF-8 support in 8bit TeX engines.

This greatly simplifies using these characters in expansion only contexts, keyval processing, but in general any `e` or `x` arguments from L3 code wheer `\protect` is not normally set up.

At the same time separate calls to `\string` on each byte are replaced by a singgle call to `\detokenize` and similarly separate calls to `\noexpand` by a single call to `\unexpanded`.  This change ensures that the inactive bytes are reconstitued as a valid UTF-8 string in a single expansion step rather than having `\string` commands between each byte (eg after f-expansion expands just teh first bye)

Finally, for issue #762, the group around the main loopis avoided, so that the definitions can be local as for other `inputenc` encodings, not global.

# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [X] Test file(s) added
- [X] Version and date string updated in changed source files
- [X] Relevant `\changes` entries in source included
- [X] Relevant `changes.txt` updated
- [X] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

**Comment by davidcarlisle** [comment]
github-077 failing is

`\MakeLowercase{Ähh}`

which relies on `Ä` expanding to `\ddot{A}` in an `\edef`.

We may need to park this PR unless this can be fixed (using the L3 lowercase code would fix it but may be a step too far)

**Comment by josephwright** [comment]
I wonder if we might update the code comments in `ltfinal`, otherwise looks good to me.

File: base/doc/ltnews35.tex [hunk 1]
```diff
@@ -323,6 +323,22 @@ \subsection{???}
 
 \section{Code improvements}
 
+\subsection{\cs{protected} UTF-8 character definitions}
+The characters defined via \file{utf8.def} are now defined as \cs{protected}
+macros. This makes them safe to use in expansion contexts where the
+classic \cs{protect} mechanism is not enabled, notably L3 programming
+layer \texttt{e} and \texttt{x} arguments.
+
```
Comment (FrankMittelbach) [inline base/doc/ltnews35.tex:331]: and file name parsing

File: base/doc/ltnews35.tex [hunk 2]
```diff
@@ -323,6 +323,22 @@ \subsection{???}
...
+Related to this change \cs{MakeUppercase} and \cs{MakeLowercase} have
+been updated to use the Unicode-aware case changing functions
+\cs{text\string_lowercase:n} in place of the \TeX-primitive \cs{lowercase}.
+A similar chaneg will be made in the \pkg{textcase} package.
+
```
Comment (FrankMittelbach) [inline base/doc/ltnews35.tex:336]: change

File: base/ltfinal.dtx [hunk 3]
```diff
@@ -1051,28 +1055,35 @@
...
 %    long-term solution is to use all-caps fonts, but these aren't
 %    generally available.
+%
+% Two wrappers around the L3 case changing functions.
+% |\protected| to make them mostly safe as replacemenst for |luppercase|
+% and |\lowercase|.  
 %    \begin{macrocode}
+\ExplSyntaxOn
+\protected\def\@text@lowercase{\text_lowercase:n}
+\protected\def\@text@uppercase{\text_uppercase:n}
```
Comment (FrankMittelbach) [inline base/ltfinal.dtx:1065]: we have a different convention here `\@expl@...@@n` I suggest you follow that.

**Review by FrankMittelbach — CHANGES_REQUESTED** [review]
perhaps also should add the firstaid for german.sty in firstaid (unless you have by now come up with a solution how to avoid it)

File: base/ltfinal.dtx (hunk 3 above)
Comment (davidcarlisle) [reply base/ltfinal.dtx:1065]: yes I wondered about that but thought the expl ones were mostly more internal  but happy to change.

**Comment by davidcarlisle** [comment]
> perhaps also should add the firstaid for german.sty in firstaid (unless you have by now come up with a solution how to avoid it)

yes done (and german.sty removed from testfiles/helpers) although I seem to have broken some tests doing that, will look tomorrow.

File: base/doc/ltnews35.tex (hunk 1 above)
Comment (davidcarlisle) [reply base/doc/ltnews35.tex:331]: actually filename parsing is already protcted via ifcsname but edef, e  and x have nothing without \protected

File: base/doc/ltnews35.tex (hunk 2 above)
Comment (davidcarlisle) [reply base/doc/ltnews35.tex:336]: done

**Pull request merged**